Sources/SDFCore/MacroInitializerRewriter.swift

```diff
@@ -142,20 +144,64 @@ private final class MacroInitializerRewriter: SyntaxRewriter {
         let infixOperand = InfixOperatorExprSyntax(foldedNode),
         infixOperand.leftOperand.trimmed.description.starts(with: "self.")
       {
-        assignments.append(
-          Assignment(infixOperand.leftOperand.description, statement))
+        let assignment = Assignment(
+          leftOperand: infixOperand.leftOperand.description,
+          item: statement,
+          group: .selfAssignments)
+        groupedAssignments.insert(assignment)
       } else {
-        otherStatements.append(statement)
+        let assignment = Assignment(
+          leftOperand: "",
+          item: statement,
+          group: .otherAssignments)
+        groupedAssignments.insert(assignment)
       }
     }
 
-    let sortedStatements: [CodeBlockItemSyntax] =
-      assignments
-      .sorted { $0.leftOperand < $1.leftOperand }
-      .map { $0.item }
+    let codeBlockItemSyntaxArray = groupedAssignments.toCodeBlockSyntasItemArray()
 
-    let codeBlockListSyntax = CodeBlockItemListSyntax(
-      sortedStatements + otherStatements)
+    let codeBlockListSyntax = CodeBlockItemListSyntax(codeBlockItemSyntaxArray)
     return body.with(\.statements, codeBlockListSyntax)
   }
 }
+
+private struct GroupedAssignments {
+  private var lastInsertedType: AssignmentGroupType = .none
+  private(set) var assignments: [[Assignment]] = []
+
+  mutating func insert(_ assignment: Assignment) {
+    switch lastInsertedType {
+    case .none:
+      assignments.append([assignment])
+    default:
+      let currentArray: [Assignment]
+      if lastInsertedType == assignment.group {
+        currentArray = assignments.removeLast() + [assignment]
+      } else {
+        currentArray = [assignment]
+      }
+
+      assignments.append(currentArray)
+    }
+
+    lastInsertedType = assignment.group
+  }
+
+  func toCodeBlockSyntasItemArray() -> [CodeBlockItemSyntax] {
```

Comment:
Note that this is not super performant, and I'm not entirely happy with iterating over all the statements to add. them to this structure and then going through it and sorting again. Not sure if we can improve a ton in perf here though, but I'd love your thoughts.